common/static/common/js/discussion/discussion_router.js

```diff
@@ -102,11 +108,43 @@
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self, callback;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (!this.thread){
+                    callback = function(thread) {
+                        self.thread = thread;
+                        self.renderThreadView();
+                    };
+                    this.retrieveSingleThread(forum_name, thread_id, callback);
```

Comment:
Tiny nit (you can ignore this whole comment if you want): I find this easier to follow when you ditch the intermediate variable and move the code into the order in which it gets called:

```
this.retrieveSingleThread(forum_name, thread_id, function(thread) {
     self.thread = thread;
     self.renderThreadView();
});
```

Or, if you wanted, instead of accepting a `callback` argument, you could make `retrieveSingleThread` so it creates returns `$.Deferred()`, and calls `deferred.resolve(new Thread(data['content']))` within the ajax `success` handler. Then you can do `this.retrieveSingleThread(forum_name, thread_id).done(function(thread) { ... });` which is even more clear.